docs/bigtable-cluster-api.rst

```diff
@@ -0,0 +1,175 @@
+
+If you want a comprehensive list of all existing clusters, make a
+`ListClusters`_ API request with
+:meth:`Client.list_clusters() <gcloud.bigtable.client.Client.list_clusters>`:
+
+.. code:: python
+
+    clusters = client.list_clusters()
+
+List Zones
+----------
+
+If you aren't sure which ``zone`` to create a cluster in, find out
+which zones your project has access to with a `ListZones`_ API request
+with :meth:`Client.list_zones() <gcloud.bigtable.client.Client.list_zones>`:
+
+.. code:: python
+
+    zones = client.list_zones()
+
+You can choose a :class:`string <str>` from among the result to pass to
+the :class:`Cluster <gcloud.bigtable.cluster.Cluster>` constructor.
+
+As of right now, the available zones are
+
+.. code:: python
+
+    >>> zones
+    [u'asia-east1-b', u'europe-west1-c', u'us-central1-c', u'us-central1-b']
+
+Cluster Factory
+---------------
+
+To create a :class:`Cluster <gcloud.bigtable.cluster.Cluster>` object:
+
+.. code:: python
+
+    cluster = client.cluster(zone, cluster_id,
+                             display_name=display_name,
+                             serve_nodes=3)
+
+Both ``display_name`` and ``serve_nodes`` are optional. When not provided,
+``display_name`` defaults to the ``cluster_id`` value and ``serve_nodes``
+defaults to the minimum allowed: ``3``.
+
+Even if this :class:`Cluster <gcloud.bigtable.cluster.Cluster>` already
+has been created with the API, you'll want this object to use as a
+parent of a :class:`Table <gcloud.bigtable.table.Table>` just as the
+:class:`Client <gcloud.bigtable.client.Client>` is used as the parent of
+a :class:`Cluster <gcloud.bigtable.cluster.Cluster>`.
+
+Create a new Cluster
+--------------------
+
+After creating the cluster object, make a `CreateCluster`_ API request
+with :meth:`create() <gcloud.bigtable.cluster.Cluster.create>`:
+
+.. code:: python
+
+    cluster.display_name = 'My very own cluster'
+    cluster.create()
+
+If you would like more than the minimum number of nodes (``3``) in your cluster:
+
+.. code:: python
+
+    cluster.serve_nodes = 10
+    cluster.create()
+
+Check on Current Operation
+--------------------------
+
+.. note::
+
+    When modifying a cluster (via a `CreateCluster`_, `UpdateCluster`_ or
+    `UndeleteCluster`_ request), the Bigtable API will return a long-running
+    `Operation`_. This will be stored on the object after each of
+    :meth:`create() <gcloud.bigtable.cluster.Cluster.create>`,
+    :meth:`update() <gcloud.bigtable.cluster.Cluster.update>` and
+    :meth:`undelete() <gcloud.bigtable.cluster.Cluster.undelete>` are called.
+
+You can check if a long-running operation (for a
+:meth:`create() <gcloud.bigtable.cluster.Cluster.create>`,
+:meth:`update() <gcloud.bigtable.cluster.Cluster.update>` or
+:meth:`undelete() <gcloud.bigtable.cluster.Cluster.undelete>`) has finished
+by making a `GetOperation`_ request with
+:meth:`Operation.finished() <gcloud.bigtable.cluster.Operation.finished>`:
+
+.. code:: python
```

Comment:
feels like there should be an actual pattern here... i sense a while loop somewhere...? this is great demo code, but doesnt seem to show what the recommended way is to "wait" on the operation or async deal with it. what is the recommended way ?  `while not operation.finished():` ?